Add Jest and Create First Tests

gilhsu opened the pull request:
Tests created for `authSlice` and `patientSlice`

Requires `redux-toolkit` merge

foxtree commented:
should we add a `.github/workflows/unit-tests.yml` (or somesuch) to  this PR, and run the Jest tests in CI?

foxtree commented on src/features/auth/authSlice.js:
```diff
@@ -1,11 +1,13 @@
   reducers: {
     setAuth: (state, action) => ({ ...state, authResult: action.payload }),
-    clearAuth: () => ({ authResult: null }),
+    clearAuth: () => (initialState),
```
👍

gilhsu commented:
> should we add a `.github/workflows/unit-tests.yml` (or somesuch) to this PR, and run the Jest tests in CI?

added this to our CI. Also updated other workflows to `setup-node@v2`
